packages/apollo-gateway/src/index.ts

```diff
@@ -428,21 +442,28 @@ export class ApolloGateway implements GraphQLService {
     };
   }
 
-  private startPollingServices() {
-    if (this.pollingTimer) clearInterval(this.pollingTimer);
+  private async pollServices() {
+    if (this.pollingTimer) clearTimeout(this.pollingTimer);
 
-    this.pollingTimer = setInterval(async () => {
-      try {
-        await this.updateComposition();
-      } catch (err) {
-        this.logger.error(err && err.message || err);
-      }
-    }, this.experimental_pollInterval || 10000);
+    try {
+      await this.updateComposition();
+    } catch (err) {
+      this.logger.error(err && err.message || err);
+    }
+
+    // Sleep for the specified pollInterval before kicking off another round of polling
+    await new Promise(res => {
+      this.pollingTimer = setTimeout(
+        res,
+        this.experimental_pollInterval || 10000,
+      );
+      // Prevent the Node.js event loop from remaining active (and preventing,
+      // e.g. process shutdown) by calling `unref` on the `Timeout`.  For more
+      // information, see https://nodejs.org/api/timers.html#timers_timeout_unref.
+      this.pollingTimer?.unref();
```

Comment:
Substantially less critical to have this now that it's not a (forever) interval, but probably worth keeping.